v2/client.go

```diff
@@ -96,9 +103,21 @@ func (c *Client) init() {
 		if c.UserAgent == "" {
 			c.UserAgent = defaultUserAgent
 		}
+		if c.OAuthConfig != nil {
+			oauthConfig := clientcredentials.Config{
+				ClientID:     c.OAuthConfig.ClientID,
+				ClientSecret: c.OAuthConfig.ClientSecret,
+				TokenURL:     c.BaseURL.ResolveReference(oauthRelTokenURL).String(),
+				Scopes:       c.OAuthConfig.Scopes,
+			}
+			// use context.Background() here, since this is used to refresh the token in the future
+			c.HTTP = oauthConfig.Client(context.Background())
+			c.HTTP.Timeout = defaultHttpClientTimeout
+		}
 		if c.HTTP == nil {
```

Comment:
and if both HTTP and OAuthConfig are specified? gets messy.

I'd just remove all this OAuth API entirely and move it to an example, showing how to set the HTTP with clientcredentials.Config.Client.

Reply:
I like that.

Reply:
Ergh, with the benefit of sleeping on it, I don't love this either. Stay tuned.

Reply:
Okay, I just created an OAuthConfig type that produces HTTP clients for use in the Client. Keeps the Client type tidy, but still provides reusable OAuth authentication support.